Subject: [PATCH] BillingCycleSettings: Fix crash case for the data usage warning

After entering a large number into the dialogue, some
digits will be converted to "E". "E" is not a valid
number, so it causes a crash.

To fix this crash, limit the input length to 7
characters which guarantees that no "E" will be
added.

diff --git a/src/com/android/settings/datausage/BillingCycleSettings.java b/src/com/android/settings/datausage/BillingCycleSettings.java
index 115939c..0cd83d5 100644
--- a/src/com/android/settings/datausage/BillingCycleSettings.java
+++ b/src/com/android/settings/datausage/BillingCycleSettings.java
@@ -29,6 +29,7 @@
 import android.support.v7.preference.Preference;
 import android.text.format.Formatter;
 import android.text.format.Time;
+import android.text.InputFilter;
 import android.util.Log;
 import android.view.LayoutInflater;
 import android.view.View;
@@ -276,6 +277,8 @@
 
                 type.setSelection(0);
             }
+            // Limit input length
+            bytesPicker.setFilters(new InputFilter[] { new InputFilter.LengthFilter(7) });
         }
 
         private String formatText(float v) {
